framework-crates/objc2-metal/translation-config.toml

```diff
@@ -33,864 +33,138 @@ enum.MTLTensorDataType.use-value = true
-# newTextureViewWithPixelFormat:textureType:levels:slices:
-# newTextureViewWithPixelFormat:textureType:levels:slices:swizzle:
-
-class.MTLBufferLayoutDescriptor.methods.stride.unsafe = false
-class.MTLBufferLayoutDescriptor.methods."setStride:".unsafe = false
-class.MTLBufferLayoutDescriptor.methods.stepFunction.unsafe = false
-class.MTLBufferLayoutDescriptor.methods."setStepFunction:".unsafe = false
-class.MTLBufferLayoutDescriptor.methods.stepRate.unsafe = false
-class.MTLBufferLayoutDescriptor.methods."setStepRate:".unsafe = false
-
-class.MTLAttributeDescriptor.methods.format.unsafe = false
-class.MTLAttributeDescriptor.methods."setFormat:".unsafe = false
-class.MTLAttributeDescriptor.methods.offset.unsafe = false
-class.MTLAttributeDescriptor.methods."setOffset:".unsafe = false
-class.MTLAttributeDescriptor.methods.bufferIndex.unsafe = false
-# TODO: Verify that index out-of-bounds is sound.
-# setBufferIndex:
-
-class.MTLVertexBufferLayoutDescriptor.methods.init.unsafe = false
-class.MTLVertexBufferLayoutDescriptor.methods.new.unsafe = false
-class.MTLVertexBufferLayoutDescriptor.methods.stride.unsafe = false
-# setStride: must be a multiple of 4
-class.MTLVertexBufferLayoutDescriptor.methods.stepFunction.unsafe = false
-class.MTLVertexBufferLayoutDescriptor.methods.stepRate.unsafe = false
-# setStepFunction: and setStepRate: must be done in lockstep
+# SAFETY: `-ffast-math` is generally frowned upon because it enables
+# optimizations that could theoretically optimize math operations away to
+# nothing. Metal enables this by default, so we must assume that their
+# configuration of it is sound (?), and that we don't need to mark compilation
+# as unsafe.
+# class.MTLCompileOptions.methods."setMathMode:".unsafe = false
+# class.MTLCompileOptions.methods."setFastMathEnabled:".unsafe = false
+
+# Using the resource's contents in a memory-safe manner is very difficult
+# after this is called.
+protocol.MTLResource.methods.makeAliasable.unsafe = true
+
+# Using `MTLHazardTrackingModeUntracked` requires extra synchronization.
+class.MTLTensorDescriptor.methods."setHazardTrackingMode:".unsafe = true
+class.MTLTextureDescriptor.methods."setHazardTrackingMode:".unsafe = true
+# TODO(breaking): Mark this as unsafe.
+# TODO: MTLHeap should maybe be unsafe by default, since it has untracked
+# by default?
+class.MTLHeapDescriptor.methods."setHazardTrackingMode:".unsafe = false
 
-class.MTLVertexAttributeDescriptor.methods.init.unsafe = false
-class.MTLVertexAttributeDescriptor.methods.new.unsafe = false
-class.MTLVertexAttributeDescriptor.methods.format.unsafe = false
-class.MTLVertexAttributeDescriptor.methods."setFormat:".unsafe = false
-class.MTLVertexAttributeDescriptor.methods.offset.unsafe = false
-# setOffset: must be a multiple of 4
-class.MTLVertexAttributeDescriptor.methods.bufferIndex.unsafe = false
-# TODO: Verify that index out-of-bounds is sound.
-# setBufferIndex:
+# TODO.
+class.MTLHeapDescriptor.methods."setType:".unsafe = true
```

Comment:
This one was safe, now it is not; what was your plan for it? It might be a breaking change, at least it'll break https://github.com/Traverse-Research/gpu-allocator if released as a patch.